cmake/FindCaDiCaL.cmake

```diff
@@ -21,18 +21,32 @@ include(deps-helper)
 find_path(CaDiCaL_INCLUDE_DIR NAMES cadical.hpp)
 find_library(CaDiCaL_LIBRARIES NAMES cadical)
 
+set(CaDiCaL_MINIMAL_VERSION "1.6.0")
+
 set(CaDiCaL_FOUND_SYSTEM FALSE)
 if(CaDiCaL_INCLUDE_DIR AND CaDiCaL_LIBRARIES)
   set(CaDiCaL_FOUND_SYSTEM TRUE)
 
-  # Unfortunately it is not part of the headers
-  find_program(CaDiCaL_BINARY NAMES cadical)
-  if(CaDiCaL_BINARY)
-    execute_process(
-      COMMAND ${CaDiCaL_BINARY} --version OUTPUT_VARIABLE CaDiCaL_VERSION
-    )
-  else()
-    set(CaDiCaL_VERSION "")
+  # Try to compile our version file
+  try_run(RUN_RESULT_VAR
+      COMPILE_RESULT_VAR
+      ${CMAKE_BINARY_DIR}
+      ${PROJECT_SOURCE_DIR}/cmake/CaDiCaL_version.cpp
+      LINK_LIBRARIES ${CaDiCaL_LIBRARIES}
+      RUN_OUTPUT_VARIABLE  CaDiCaL_VERSION
+  )
+
+  # If this failed to compile or run, we have bigger issues
+  if (NOT ${RUN_RESULT_VAR} EQUAL 0 OR NOT ${COMPILE_RESULT_VAR})
+    message(FATAL_ERROR "Failed to detect CaDiCaL version")
+  endif()
+
+  # Remove any whitespace
+  string(STRIP ${CaDiCaL_VERSION} CaDiCaL_VERSION)
+
+  # Let's check if the version is sufficient
+  if(CaDiCaL_VERSION VERSION_LESS ${CaDiCaL_MINIMAL_VERSION})
+    message(FATAL_ERROR "CaDiCaL ${CaDiCaL_VERSION} is too old; minimum required ${CaDiCaL_MINIMAL_VERSION}")
```

Comment:
Probably we want this to revert to `if(NOT CaDiCaL_FOUND_SYSTEM)` if this fails, right?

That is: system version is too old -> okay, cvc5 will build its own

Reply:
Yes.

Reply:
All implemented!